### ISSUE TITLE
Cannot connect to Redshift with IAM credentials

### ISSUE THREAD

**Issue by meih** [aws/amazon-redshift-python-driver#19]
## Driver version

2.0.874

## Redshift version

Redshift 1.0.23412

## Client Operating System

Mac OS 10.15.7

## Python version

3.9.1

## Table schema

N/A

## Problem description

1. Expected behaviour:
I can connect to Redshift using IAM credentials, with the following values specified (as written in the README):

```
conn = redshift_connector.connect(
    iam=True,
    database='my_databse',
    db_user='my_db_user',
    password='',
    user='',
    cluster_identifier='my_cluster_identifier',
    access_key_id="my_aws_access_key_id",
    secret_access_key="my_aws_secret_access_key",
    session_token="my_aws_session_token",
    region="my_region"
 )
```

2. Actual behaviour:
It failed with the following error message, although I'm setting `iam=True` .

3. Error message/stack trace:
redshift_connector.error.InterfaceError: Invalid connection property setting. user, password, host, database, port, and password  are required.

4. Any other details that can be helpful:
When I tried user/password authentication with the following values specified, it was successful.

```
conn = redshift_connector.connect(
    host='my_cluster.my_region.redshift.amazonaws.com',
    database='my_database',
    user='my_username',
    password='my_password'
 )
```

## Python Driver trace logs

```
Traceback (most recent call last):
  File "~/Documents/github/sandbox/RedshiftConnectorTest/test.py", line 26, in <module>
    conn = redshift_connector.connect(
  File "~/.pyenv/versions/3.9.1/lib/python3.9/site-packages/redshift_connector/__init__.py", line 126, in connect
    IamHelper.set_iam_properties(
  File "~/.pyenv/versions/3.9.1/lib/python3.9/site-packages/redshift_connector/iam_helper.py", line 188, in set_iam_properties
    raise InterfaceError(
redshift_connector.error.InterfaceError: Invalid connection property setting. user, password, host, database, port, and password  are required.
```

## Reproduction code

N/A

**Comment by iggarish** [comment]
Thanks for reporting issue. We will look into the issue and get back to you.

**Comment by Brooke-white** [comment]
Hi @meih ,

Thank you for your patience. We plan to include a fix for this issue in the next release of `redshift_connector`, which is next week.

**Comment by meih** [comment]
@Brooke-white 
Sounds great, thank you!

**Comment by Brooke-white** [comment]
Hi @meih ,

We've released [v2.0.875](https://github.com/aws/amazon-redshift-python-driver/blob/master/CHANGELOG.md#v20875-2021-03-08) of `redshift_connector`, which includes a fix for this issue. I'll be closing this issue, but feel free to re-open if you continue experiencing issues with IAM credentials.
